doc/varnish-configuration.rst

```diff
@@ -36,111 +36,162 @@ will be used throughout the Varnish examples on this page.
+its variants.
 
-Subroutines are provided in ``fos_refresh.vcl``
+Subroutines are provided in ``resources/config/varnish-[version]/fos_refresh.vcl``.
+To enable support, add the following to ``your_varnish.vcl``:
 
-To enable support add the following to ``your_varnish.vcl``:
+.. configuration-block::
 
-.. code-block:: varnish3
+    .. code-block:: varnish4
 
-    include "path-to-config/varnish-[version]/fos_refresh.vcl";
+        include "path-to-config/varnish-4/fos_refresh.vcl";
 
-.. configuration-block::
-    .. literalinclude:: ../tests/Functional/Fixtures/varnish-4/fos.vcl
-        :language: varnish4
-        :lines: 17,20-22
+        sub vcl_recv {
+            call fos_refresh_recv;
+        }
 
-    .. literalinclude:: ../tests/Functional/Fixtures/varnish-3/fos.vcl
-        :language: varnish3
-        :lines: 15,18-20
+    .. code-block:: varnish3
+
+        include "path-to-config/varnish-3/fos_refresh.vcl";
+
+        sub vcl_recv {
+            call fos_refresh_recv;
+        }
+
+Read more on `forcing a refresh`_ in the Varnish documentation (for Varnish 3,
+see `refreshing for Varnish 3`_).
 
 Ban
 ~~~
 
-To configure Varnish for `handling BAN requests <https://www.varnish-software.com/static/book/Cache_invalidation.html#banning>`_:
-
-Subroutines are provided in ``fos_ban.vcl``
+Banning invalidates whole groups of cached entries with regular expressions.
 
+Subroutines are provided in ``resources/config/varnish-[version]/fos_ban.vcl``
 To enable support add the following to ``your_varnish.vcl``:
 
-.. code-block:: varnish3
+.. configuration-block::
 
-    include "path-to-config/varnish-[version]/fos_ban.vcl";
+    .. code-block:: varnish4
 
-.. configuration-block::
-    .. literalinclude:: ../tests/Functional/Fixtures/varnish-4/fos.vcl
-        :language: varnish4
-        :lines: 17-18,21-29
+        include "path-to-config/varnish-4/fos_ban.vcl";
 
-    .. literalinclude:: ../tests/Functional/Fixtures/varnish-3/fos.vcl
-        :language: varnish3
-        :lines: 15-16,19-24,33-35
+        sub vcl_recv {
+            call fos_ban_recv;
+        }
 
-Varnish contains a `ban lurker`_ that crawls the content to eventually throw out banned data even when it’s not requested by any client.
+        sub vcl_backend_response {
+            call fos_ban_backend_response;
+        }
 
-.. _ban lurker: https://www.varnish-software.com/blog/ban-lurker
+        sub vcl_deliver {
+            call fos_ban_deliver;
+        }
+
+    .. code-block:: varnish3
+
+        include "path-to-config/varnish-3/fos_ban.vcl";
+
+        sub vcl_recv {
+            call fos_ban_recv;
+        }
+
+        sub vcl_fetch {
+            call fos_ban_fetch;
+        }
+
+        sub vcl_deliver {
+            call fos_ban_deliver;
+        }
+
+Varnish contains a `ban lurker`_ that crawls the content to eventually throw
+out banned data even when it’s not requested by any client. Read more on
```

Comment:
The reference to the ban lurker is a bit out of the blue, here. Perhaps we should add that the FOSHttpCache VCL subroutines set ban lurker-friendly headers (in other words, is compatible with Varnish's ban lurker)?